contracts/examples/DXswapRelayer.sol

```diff
@@ -0,0 +1,325 @@
+            (uint256 amountToken, uint256 amountETH, uint256 liquidity) = IDXswapRouter(dxSwapRouter).addLiquidityETH{
+                value: value
+            }(token, amount, minToken, minETH, dxdaoAvatar, block.timestamp);
+            emit Pool(address(0), token, amountETH, amountToken, minETH, minToken, liquidity);
+        }
+    }
+
+    function getPriceOracleFactory(
+        address tokenA,
+        address tokenB,
+        uint256 minReserveA,
+        uint256 minReserveB
+    ) internal view returns (address factory) {
+        (uint256 dxSwapReserveA, uint256 dxSwapReserveB) = DXswapLibrary.getReserves(dxSwapFactory, tokenA, tokenB);
+        (uint256 uniswapReserveA, uint256 uniswapReserveB) = DXswapLibrary.getReserves(uniswapFactory, tokenA, tokenB);
+        require(
+            (dxSwapReserveA >= minReserveA && dxSwapReserveB >= minReserveB) ||
+                (uniswapReserveA >= minReserveA && uniswapReserveB >= minReserveB),
+            'DXliquidityRelay: INSUFFICIENT_RESERVE'
+        );
+        if (dxSwapReserveA >= minReserveA && dxSwapReserveB >= minReserveB) {
+            factory = dxSwapFactory;
+        } else {
+            factory = uniswapFactory;
+        }
+    }
+
+    function getPoolStake(
+        address factory,
+        address tokenA,
+        address tokenB,
+        uint256 amountA,
+        uint256 amountB
+    ) internal view returns (uint256 poolStake) {
+        (uint256 reserveA, uint256 reserveB) = DXswapLibrary.getReserves(factory, tokenA, tokenB);
+        poolStake = (amountA.add(amountB)) / ((reserveA.add(reserveB))).mul(10000);
+    }
+
+    function consultOracleParameters(
+        IDXswapFactory factory,
+        address tokenA,
+        address tokenB,
+        uint256 amountA,
+        uint256 amountB,
+        uint256 maxWindows
+    ) internal view returns (uint256 windowSize, uint8 granularity) {
+        uint256 poolStake = getPoolStake(address(factory), tokenA, tokenB, amountA, amountB);
+        // poolStake: 0.1% = 10; 1=100; 10% = 1000;
+        uint256 windows;
+        if(poolStake < 10) {
+          windows = 3;
+        } else if (poolStake >= 10 && poolStake < 25){
+          windows = 4;
+        } else if (poolStake >= 25 && poolStake < 50){
+          windows = 5;
+        } else if (poolStake >= 50 && poolStake < 100){
+          windows = 6;
+        } else if (poolStake >= 100 && poolStake < 300){
+          windows = 6;
+        } else {
+          windows = 10;
+        }
+        windows = windows <= maxWindows ? windows : maxWindows;
+        windowSize = windows.mul(baseWindowSize);
+        granularity = uint8(windowSize / baseWindowSize);
+    }
+
+    function ERC20Withdraw(address token, uint256 amount) public {
+        require(msg.sender == dxdaoAvatar, 'DXliquidityRelay: CALLER_NOT_DXDAO');
+        TransferHelper.safeTransfer(token, dxdaoAvatar, amount);
+    }
+
+    function EthWithdraw(uint256 amount) public {
```

Comment:
We never used Eth, we always used ETH in uppercase in the code, you might want to change this function name.